h2o-automl/src/main/java/ai/h2o/automl/ModelingStep.java

```diff
@@ -105,6 +106,45 @@
         return builder.trainModelOnH2ONode();
     }
 
+
+    public boolean supportsDistribution(DistributionFamily distributionFamily) {
+        throw new UnsupportedOperationException();
+    }
+
+    public boolean validParameters(Model.Parameters parms, String[] fields) {
+        try {
+            Model.Parameters params = parms.clone();
+            // some algos check if distribution has proper _nclass(es) so we need to set training frame and response etc
+            setCommonModelBuilderParams(params);
+            ModelBuilder mb = ModelBuilder.make(params);
+            mb.init(false);
+            return Arrays.stream(fields)
+                    .allMatch((field) ->
+                            mb.getMessagesByFieldAndSeverity(field, Log.ERRR).length == 0);
+        } catch (H2OIllegalArgumentException e) {
+            return false;
+        }
+    }
+
+    protected void setDistributionParameters(Model.Parameters parms) {
```

Comment:
imo, this should be applied directly for all models in `ModelingStep.ModelStep.trainModel`, and `ModelingStep.GridStep.hyperparameterSearch` as that's where user input parameters are applied, not in each individual step.